### PR TITLE
Ensure prerequisites for deleting operator and account role policies

### PR DESCRIPTION
**Pull request by oriAdler**
1. Check if the policy is attached to an entity.
2. Delete all policy versions.

Related: [SDA-6747](https://issues.redhat.com/browse/SDA-6747)

Delete a policy with multiple versions:
image

Skip policy deletion if it has attached entities:
image

### CODE REVIEW

**Comment by oriAdler** [comment]
@tbrisker @ydayagi @hunterkepley PTAL

**Review by hunterkepley — COMMENTED** [review]
lgtm

File: pkg/aws/policies.go [hunk 1]
```diff
@@ -889,10 +879,37 @@ func (c *awsClient) DeleteInlineRolePolicies(role string) error {
...
+	if err != nil {
+		return false, err
+	}
+
+	isAttached := false
+	if *policyOutput.Policy.AttachmentCount > 0 {
+		isAttached = true
+	}
+
+	return isAttached, nil
```
Comment (tbrisker) [inline pkg/aws/policies.go:893]:
```suggestion
	if *policyOutput.Policy.AttachmentCount > 0 {
		return true, nil
	}

	return false, nil
```

File: pkg/aws/policies.go [hunk 2]
```diff
@@ -889,10 +879,37 @@ func (c *awsClient) DeleteInlineRolePolicies(role string) error {
...
+}
+
 func (c *awsClient) deletePolicies(policies []string) (*iam.DeletePolicyOutput, error) {
 	var output *iam.DeletePolicyOutput
-	var err error
+
 	for i := range policies {
```
Comment (tbrisker) [inline pkg/aws/policies.go:899]: np, why not:
```suggestion
	for _, policy := range policies {
```

**Review by tbrisker — COMMENTED** [review]
looks good in general, a couple of minor comments inline

File: pkg/aws/policies.go (hunk 1 above)
Comment (oriAdler) [reply pkg/aws/policies.go]: Changed, thanks.

File: pkg/aws/policies.go (hunk 2 above)
Comment (oriAdler) [reply pkg/aws/policies.go]: Changed.

Comment (tbrisker) [reply pkg/aws/policies.go]: ah, I guess that's why it was using `i`, sorry for the noise:
```
 pkg/aws/policies.go:912:76: G601: Implicit memory aliasing in for loop. (gosec)
		output, err = c.iamClient.DeletePolicy(&iam.DeletePolicyInput{PolicyArn: &policy})
		                                                                         ^
make: *** [lint] Error 1

File: pkg/aws/policies.go (hunk 1 above)
Comment (oriAdler) [reply pkg/aws/policies.go]: Changed, thanks.

**Pull request merged**